sklearn/cluster/affinity_propagation_.py

```diff
@@ -90,6 +106,20 @@ def affinity_propagation(S, preference=None, convergence_iter=15, max_iter=200,
     if damping < 0.5 or damping >= 1:
         raise ValueError('damping must be >= 0.5 and < 1')
 
+    if n_samples == 1 or equal_similarities_and_preferences(S, preference):
+        # It makes no sense to run the algorithm in this case, so return 1 or
+        # n_samples clusters, depending on preferences
+        warnings.warn("All samples have mutually equal similarities, "
+                      "returning arbitrary cluster center(s).")
```

Comment:
should this be a `ConvergenceWarning`? I suppose `UserWarning` is fine.

Reply:
It is a `UserWarning` right now